providers/common/ciphers/cipher_gcm_hw.c

```diff
@@ -30,14 +30,24 @@ int gcm_cipher_update(PROV_GCM_CTX *ctx, const unsigned char *in,
 #if defined(AES_GCM_ASM)
             size_t bulk = 0;
 
+# if defined(AES_PMULL_CAPABLE)
```

Comment:
I think it may be worth giving this code its own update() so we dont end up with a horrible mess of #ifdef starting to propagate in here.. What do you think?

Reply:
I agree with you. I'm thinking if we can share the update() function while define length macro and encrypt function macro  for different architectures.